e2e/README.md

```diff
@@ -0,0 +1,66 @@
+# End-to-End (E2E) Tests
+
+## Testing Philosophy
+
+The E2E tests in this directory are designed to test Trivy's functionality in realistic environments with **external dependencies and network connectivity**. These tests complement unit tests and integration tests by focusing on scenarios that require real external resources.
+
+### What E2E Tests Should Cover
+
+E2E tests should focus on functionality that involves:
+- **External network connections** (downloading container images, vulnerability databases)
+- **External service dependencies** (Docker daemon, registry access, proxy servers)
+- **Real-world scenarios** that cannot be easily mocked or simulated
+- **Cross-component integration** involving external systems
+
+### What E2E Tests Should NOT Cover
+
+E2E tests should **avoid** detailed assertions and comprehensive validation:
+- **Detailed JSON output validation** - this should be covered by integration tests
+- **Comprehensive vulnerability detection** - this should be covered by unit tests
+- **Complex result comparison** - basic functionality verification is sufficient
+- **Edge cases and error conditions** - these should be covered by unit tests
+
+### Testing Approach
+
+- **Minimal assertions**: Focus on basic functionality rather than detailed output validation
+- **External dependencies**: Use real registries, databases, and services where practical
+- **Environment isolation**: Each test should use isolated cache and working directories
+- **Golden files**: Use -update flag for maintainable output comparison
+- **Conditional execution**: Tests should validate required dependencies during setup
+
+### Dependencies
+
+- **Docker**: Required for local image scanning tests
+- **Internet access**: Required for downloading images and databases
+
+### Test Execution
+
+The E2E tests execute the trivy binary from `$GOPATH/bin`. When you run `mage test:e2e`, it automatically:
```

Comment:
IIUC now path to Trivy binary is `<temp_dir>/trivy(.exe)`:
https://github.com/aquasecurity/trivy/blob/0b8ca0b19907bb6f17c1e4e155c84a3212b0a82d/e2e/e2e_test.go#L32-L41

Reply:
Oh, I forgot to update the document. Thanks.
https://github.com/aquasecurity/trivy/pull/9231/commits/c37a67d7d07252368154220ad198052ddddf101c